src/FRP/Event/Time.purs

```diff
@@ -15,3 +17,22 @@ foreign import animationFrame :: Event Unit
 
 -- | Create an event which reports the current time in milliseconds since the epoch.
 foreign import withTime :: forall a. Event a -> Event { value :: a, time :: Int }
+
+-- | Similar to `sampleOn_`, except that the returned `Event` is a `Tuple` of
+-- | the sampled value _as well as_ the time since that value was pushed to the
+-- | stream (in milliseconds).
+withDelay :: forall a b. Event a -> Event b -> Event (Tuple Int a)
```

Comment:
I'd like to have a think about how we could implement this in one of the Haskell libraries.

Would you mind breaking this PR up a bit, so that we can merge things separately?